infra/src/main/kotlin/fi/fta/geoviite/infra/publication/Publication.kt

```diff
@@ -29,16 +32,39 @@ enum class PublicationTableColumn {
 data class PublicationTableItem(
     val name: String,
     val trackNumbers: List<TrackNumber>,
-    val changedKmNumbers: List<KmNumber>? = null,
+    val changedKmNumbers: String?,
     val operation: Operation,
     val publicationTime: Instant,
     val publicationUser: UserName,
     val message: String,
     val ratkoPushTime: Instant?,
+    val propChanges: List<PublicationChange<*>>,
 ) {
     val id: StringId<PublicationTableItem> = StringId(hashCode().toString())
 }
 
+data class ChangeValue<T>(
+    val oldValue: T?,
+    val newValue: T?,
+    val localizationKey: LocalizationKey? = null,
+) { constructor(oldValue: T?, newValue: T?, localizationKey: String?) : this(oldValue, newValue, localizationKey?.let(::LocalizationKey)) }
+
+data class PublicationChange<T>(
+    val propKey: PropKey,
+    val value: ChangeValue<T>,
+    val remark: PublicationChangeRemark?,
+)
```

Comment:
Tässä varsinainen tietomalli, jossa muutokset siirretään. Näistä tärkeimpänä tämä `PublicationChange<T>`, joka kuvaa yksittäisen kentän muutosta. `propKey` kuvaa kentän "otsikkoa" parametreineen, `value` sisältää vanhan ja uuden arvon mahdollisine käännösavaimineen ja `remark` sisältää huomion (taaskin mahdollisine parametrisointeineen.) Kantavina ideoina tässä on ollut se, että a) frontille lähetetään aina pelkästään muuttuneet kentät ja b) bäkkäri pureksii arvot frontille mahdollisimman pitkälle (käännökset poislukien.) Tällöin frontille ei tarvitse välittää tietoa siitä minkälaista tietomallia bäkkäriltä tuli, vaan se voi vaan näyttää ne mahdollisimman simppelisti